database/upload.go

```diff
@@ -45,15 +46,18 @@ func (db *DB) UploadByID(ctx context.Context, id primitive.ObjectID) (*Upload, e
 
 // UploadCreate registers a new upload and counts it towards the user's used
 // storage.
-func (db *DB) UploadCreate(ctx context.Context, user User, skylink Skylink) (*Upload, error) {
-	if user.ID.IsZero() {
-		return nil, errors.New("invalid user")
+func (db *DB) UploadCreate(ctx context.Context, user *User, ip string, skylink Skylink) (*Upload, error) {
+	if user == nil {
+		// If there is no user passed, we initialise it with the zero ID in
+		// order to denote an anonymous upload.
+		user = &User{ID: primitive.ObjectID{}}
```

Comment:
Similar thing as below. This seems like it could hide errors.
I would feel more comfortable with specifying a dedicated and global `AnonUser = &User{ID: primitive.ObjectID{}}` which we pass explicitly.

Reply:
Added.